src/main/java/com/atlassian/bitbucket/jenkins/internal/applink/oauth/RandomizerImpl.java

```diff
@@ -1,10 +1,85 @@
 package com.atlassian.bitbucket.jenkins.internal.applink.oauth;
 
-import org.apache.commons.lang3.RandomStringUtils;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.lang3.StringUtils;
 
+import java.security.SecureRandom;
+
+/**
+ * Implementation of {@link Randomizer} that uses the Java {@link SecureRandom} to generate random strings
+ */
 public class RandomizerImpl implements Randomizer {
```

Comment:
Can you java-doc the details as to why we simply did not rely on secure random to generate a token of a given length and why we need to have this implementation?

Reply:
But we _are_ using `SecureRandom`. You mean document why we're limiting the chars to alphanum or URL-safe?